tests/test_communication_manager.py

```diff
@@ -0,0 +1,153 @@
+    (7) Optimizer updates its weights to initialized model
+    (8) Optimizer tells Communication Manager to schedule a training job
+    (9) Communication Manager schedules training job
+    (10) Communication Manager receives DMLResult for training job
+    (11) Communication Manager gives DMLResult to Optimizer
+    (12) Optimizer updates its weights to trained weights
+    (13) Optimizer tells Communication Manager to schedule a communication job
+    (14) Communication Manager schedules communication job
+    (15) Exception raised, since communication job is not implemented
+    NOTE: No, I can't split this up. IDK why exactly but it hangs forever.
+    """
+    communication_manager = CommunicationManager()
+    scheduler = DMLScheduler(config_manager, communication_manager)
+    communication_manager.configure(scheduler)
+    new_session_event = {
+        "key": None,
+        "content": {
+            "optimizer_params": {},
+            "serialized_job": make_serialized_job()
+        }
+    }
+    communication_manager.inform(RawEventTypes.NEW_SESSION.name, new_session_event)
+    scheduler.runners_run_next_jobs()
+    scheduler.runners_run_next_jobs()
+    scheduler.runners_run_next_jobs()
+    assert len(scheduler.processed) == 2
+
+def test_communication_manager_can_be_initialized():
+    communication_manager = CommunicationManager()
+    assert communication_manager
+
+def test_communication_manager_creates_new_sessions():
+    """
+    Takes in initialized_communication_manager because the initialization is all that matters.
+    Ensures that upon receiving an initialization job, the Communication Manager will make an optimizer.
+    """
+    communication_manager = CommunicationManager()
+    scheduler = DMLScheduler(config_manager, communication_manager)
+    communication_manager.configure(scheduler)
+    serialized_job = make_serialized_job()
+    new_session_event = {
+        "key": None,
+        "content": {
+            "optimizer_params": {},
+            "serialized_job": serialized_job
+        }
+    }
+    communication_manager.inform(RawEventTypes.NEW_SESSION.name, new_session_event)
+    assert communication_manager.optimizer
+
+def test_communication_manager_fails_if_not_configured():
+    """
+    Ensures that Communication Manager won't be able to function if it's not configured
+    """
+    communication_manager = CommunicationManager()
+    serialized_job = make_serialized_job()
+    new_session_event = {
+        "key": None,
+        "content": {
+            "optimizer_params": {},
+            "serialized_job": serialized_job
+        }
+    }
+    try:
+        communication_manager.inform(RawEventTypes.NEW_SESSION.name, new_session_event)
+        raise Exception("This should have raised an exception")
+    except Exception as e:
+        assert str(e) == "Communication Manager needs to be configured first!"
+
+def test_communication_manager_can_inform_the_optimizer():
```

Comment:
Please rename to `test_communication_manager_can_inform_new_job_to_the_optimizer` or something more precise.

Reply:
done